cdap-api/src/main/java/co/cask/cdap/api/plugin/EndpointPluginContext.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.plugin;
+
+import co.cask.cdap.api.annotation.Beta;
+
+import javax.annotation.Nullable;
+
+/**
+ * Provides access to load plugin class from plugin endpoint methods.
+ */
+@Beta
+public interface EndpointPluginContext {
+  /**
+   * Instantiates and returns a plugin class.
```

Comment:
what does it mean to instantiate a plugin class?  The wording seems a little weird to me, I would expect instantiate to mean I get an object back instead of a class.  

Maybe say "Loads and returns", which is how PluginContext  words it